File: acm/Online-Judge/leetcode/code/contains-duplicate-ii.cpp
class Solution {
public:
    bool containsNearbyDuplicate(vector<int>& nums, int k) {
        int n=nums.size();
        unordered_map<int,int>pos;
        for(int i=0;i<n;i++)
            if(pos.count(nums[i])==0)
                pos[nums[i]]=i;
            else
            {
                if(i-pos[nums[i]]<=k)
                    return true;
                pos[nums[i]]=i;
            }
        return false;
    }
};

File: acm/Online-Judge/leetcode/code/contains-duplicate-ii.py
class Solution:
    # @param {integer[]} nums
    # @param {integer} k
    # @return {boolean}
    def containsNearbyDuplicate(self, nums, k):
        n=len(nums)
        pos={}
        for i in xrange(n):
            if pos.get(nums[i])==None:
                pos[nums[i]]=i
            else:
                if i-pos[nums[i]]<=k:
                    return True
                pos[nums[i]]=i
        return False